### ISSUE TITLE
Define unified data loader for 2D perception

### ISSUE THREAD

**Issue by kattoyoshi** [kattoyoshi/PytorchInfinity#6]
## Goal
- Create unified data loader for 2D perception data.
- Support
  - Segmentation id label
  - 2D bounding box
  - Classification

### Create pre-process tool to make unified gt data.
Concept
- Instead of implementing a lot of dataset class, create single unified dataset class for 2D perception (including multi-task support) to keep the dataloader implementation simple

In this issue, I will support.
- MSCoCo
- MNIST
- BDD100k

Candidate data format is dataframe.
Candidate library is
- pandas 2.0 with pyarrow
- [polars](https://github.com/pola-rs/polars)
- [torch arrow](https://github.com/pytorch/torcharrow)

(1) Create data loader with DALI
(2) Store data as dictionary of pre-defined data class
- [Segmentation](https://github.com/open-mmlab/mmsegmentation/blob/main/mmseg/structures/seg_data_sample.py)
- [2D bbox](https://github.com/open-mmlab/mmdetection/blob/main/mmdet/structures/det_data_sample.py)
- [Classification](https://github.com/open-mmlab/mmpretrain/blob/main/mmpretrain/structures/data_sample.py)
- Lane detection: Create new data class

(3) Define augmentation set using albumentation/torchvision/konia

## Future works
- Support lane detection data
